libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2229,6 +2230,70 @@ bool WaveTrack::GetOne(samplePtr buffer,
    return result;
 }
 
+std::vector<TrackSampleView> WaveTrack::GetSampleView(
+   size_t iChannel, size_t nBuffers, sampleCount start, size_t length) const
+{
+   const auto nChannels = NChannels();
+   assert(nBuffers - iChannel <= nChannels); // precondition
+   const auto pOwner = GetOwner();
+   if (!pOwner) {
+      // GetSampleView called on a WaveTrack outside a project ??
+      // Not a complication I want to address, all the more that ongoing
+      // refactoring should rid us of this call.
+      assert(false);
+      return {};
```

Comment:
But see `WaveTrack::Get`.  You can easily copy some of that to make this function more robust.

```
   const auto pOwner = GetOwner();
   if (!pOwner) {
      //! an un-owned track should have reported one channel only
      assert(nChannels == 1);
      nBuffers = std::min<size_t>(nBuffers, 1);
   }
```

Reply:
This is better than your not wholly justified `assert(false)`.  Instead it asserts a postcondition of `NBuffers()`.

Actually that's not an explicitly stated postcondition in the header, but that can and should be added easily.

Reply:
I'm addressing this reluctantly, I was hoping for no debate and time spending over this.